Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/measure.h

```diff
@@ -150,6 +151,93 @@ edge_length(typename boost::graph_traits<PolygonMesh>::edge_descriptor e,
   return edge_length(halfedge(e, pmesh), pmesh);
 }
 
+/**
+  * \ingroup measure_grp
+  * computes the squared length of an edge of a given polygon mesh.
+  * The edge is given by one of its halfedges, or the edge itself.
+  *
+  * @tparam PolygonMesh a model of `HalfedgeGraph`
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param h one halfedge of the edge to compute the squared length
+  * @param pmesh the polygon mesh to which `h` belongs
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamNBegin{vertex_point_map}
+  *     \cgalParamDescription{a property map associating points to the vertices of `pmesh`}
+  *     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+  *                    as key type and `%Point_3` as value type}
+  *     \cgalParamDefault{`boost::get(CGAL::vertex_point, pmesh)`}
+  *   \cgalParamNEnd
+  *
+ *   \cgalParamNBegin{geom_traits}
+ *     \cgalParamDescription{an instance of a geometric traits class}
+ *     \cgalParamType{a class model of `Kernel`}
+ *     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+ *     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+ *   \cgalParamNEnd
+  * \cgalNamedParamsEnd
+  *
```

Comment:
\pre h non-null ?

Reply:
Is n't that in general the case?

Reply:
You made the requirement explicit in the doc of `squared_face_area()` (or at least it was so in `face_area()`, I guess). It's just symmetry, not very important; the precondition in the actual code is more important.

Reply:
I added an assertion.